src/populateExchange.ts

```diff
@@ -133,31 +150,45 @@ interface TypeFragment {
 /** Gets typed selection sets and fragments from query */
 export const extractSelectionsFromQuery = (
   schema: GraphQLSchema,
-  query: DocumentNode
+  query: DocumentNode,
+  variables: Variables
 ) => {
   const extractedFragments: FragmentDefinitionNode[] = [];
   const newFragments: FragmentDefinitionNode[] = [];
   const typeInfo = new TypeInfo(schema);
 
+  const handleVisit = (node: OperationDefinitionNode | FieldNode) => {
+    if (node.selectionSet) {
+      const type = getTypeName(typeInfo);
+      newFragments.push({
+        kind: Kind.FRAGMENT_DEFINITION,
+        typeCondition: {
+          kind: Kind.NAMED_TYPE,
+          name: nameNode(type),
+        },
+        name: nameNode(`${type}_PopulateFragment_`),
+        selectionSet: node.selectionSet,
+      });
+    }
+  };
+
   visit(
     query,
     visitWithTypeInfo(typeInfo, {
-      Field: node => {
-        if (node.selectionSet) {
-          const type = getTypeName(typeInfo);
-          newFragments.push({
-            kind: Kind.FRAGMENT_DEFINITION,
-            typeCondition: {
-              kind: Kind.NAMED_TYPE,
-              name: nameNode(type),
-            },
-            name: nameNode(`${type}_PopulateFragment_`),
-            selectionSet: node.selectionSet,
-          });
-        }
+      OperationDefinition: {
+        leave: handleVisit,
+      },
+      Field: {
+        leave: handleVisit,
+      },
+      Variable: {
+        enter: v => {
+          const type = typeInfo.getInputType() as GraphQLInputType;
+          return astFromValue(variables[getName(v)], type);
+        },
       },
       FragmentDefinition: node => {
-        extractedFragments.push(node);
+        leave: extractedFragments.push(node);
```

Comment:
This isn’t an object so `leave` here is actually a label 😅 seems like this change can be reverted 🙂